helpers/form/class.FormContainer.php

```diff
@@ -100,6 +93,8 @@ public function __construct($data = [], $options = [])
         // initialize the elements of the form
         $this->initElements();
 
+        $this->applyAdditionalValidationRules($options[self::ADDITIONAL_VALIDATORS] ?? []);
+
```

Comment:
```suggestion
        if (!empty($options[self::ADDITIONAL_VALIDATORS])) {
            $this->applyAdditionalValidationRules($options[self::ADDITIONAL_VALIDATORS]);
        }

```

Otherwise we would be looping the form unnecessarily every time.